File: src/Python/179.LargestNumber.py
import functools

def cmp(a, b):
    return 1 if a + b < b + a else -1

class Solution:
    def largestNumber(self, nums):
        numsstr = list(map(str, nums))
        numsstr.sort(key=functools.cmp_to_key(cmp))
        if numsstr[0] == '0':
            return '0'
        return ''.join(numsstr)


if __name__ == "__main__":
    solution = Solution()
    print("102" < "210")
    print(solution.largestNumber([0, 0]))
    print(solution.largestNumber([10, 2]))
    print(solution.largestNumber([3, 30, 34, 5, 9]))

File: src/CSharp/101-200/179.LargestNumber.cs

using System;
using System.Text;
using System.Collections.Generic;

namespace PeefyLeetCode.LargestNumber
{
    public class LargerNumberComparator : IComparer<string> {
        public int Compare(string x, string y)
        {
            string order1 = x + y;
            string order2 = y + x;
            return order2.CompareTo(order1);
        }

        public override bool Equals(object obj)
        {
            return base.Equals(obj);
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }

        public override string ToString()
        {
            return base.ToString();
        }
    }

public class Solution {
    public string LargestNumber(int[] nums) {
        // Get input integers as strings.
        String[] asStrs = new String[nums.Length];
        for (int i = 0; i < nums.Length; i++) {
            asStrs[i] = nums[i].ToString();
        }

        // Sort strings according to custom comparator.
        Array.Sort(asStrs, new LargerNumberComparator());

        // If, after being sorted, the largest number is `0`, the entire number
        // is zero.
        if (asStrs[0].Equals("0")) {
            return "0";
        }

        // Build largest number from sorted array.
        StringBuilder largestNumberStr = new StringBuilder();
        foreach (var numAsStr in asStrs) {
            largestNumberStr.Append(numAsStr);
        }

        return largestNumberStr.ToString();
    }
}

}



